javaslang-benchmark/src/test/java/javaslang/idiom/TryBenchmark.java

```diff
@@ -0,0 +1,111 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.collection.Array;
+import javaslang.control.Try;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+import java.util.Random;
+
+import static javaslang.API.Case;
+import static javaslang.API.Match;
+import static javaslang.Predicates.instanceOf;
+
+/**
+ * Benchmark for Try vs try/catch.
+ */
+public class TryBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            TryThrowsCatchBenchmark.class,
+            TryNoThrowsCatchBenchmark.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+
+    @State(Scope.Benchmark)
+    public static class Base {
+
+        int random;
+
+        Integer throwsException() {
+            if ((random ^ random) == 0) throw new RuntimeException();
```

Comment:
Because in the "plain java" benchmark, if the jvm figures out one of the branches is not reachable due to a method never/always throwing an exception, the comparison is not really fair. In real life code, it would be much harder to figure out that a branch is not reachable.

Maybe I'm overreaching here :)